app_dart/lib/src/request_handlers/refresh_chromebot_status.dart

```diff
@@ -71,38 +71,36 @@ class RefreshChromebotStatus extends ApiRequestHandler<Body> {
 
   /// Update chromebot tasks statuses in datastore for [builder],
   /// based on latest [luciTasks] statuses.
-  Future<void> _updateStatus(LuciBuilder builder, String branch,
-      DatastoreService datastore, Map<String, List<LuciTask>> luciTasks) async {
+  Future<void> _updateStatus(
+      LuciBuilder builder,
+      String branch,
+      DatastoreService datastore,
+      Map<String, List<LuciTask>> luciTasksMap) async {
     final List<FullTask> datastoreTasks = await datastore
         .queryRecentTasks(taskName: builder.taskName, branch: branch)
         .toList();
-    final Set<LuciTask> updatedLuciTasks = <LuciTask>{};
 
-    /// Since [datastoreTasks] may contain new re-run builds which are not scheduled yet in luci,
-    /// there may not be a strict one-to-one mapping. Therefore we scan both [datastoreTasks]
-    /// and [luciTasks] from old to new. If matched, then update accordingly.
-    for (FullTask datastoreTask in datastoreTasks.reversed) {
-      if (!luciTasks.containsKey(datastoreTask.commit.sha)) {
-        continue;
-      }
-      for (LuciTask luciTask in luciTasks[datastoreTask.commit.sha].reversed) {
-        if (!updatedLuciTasks.contains(luciTask) &&
-            _buildNumberMatched(datastoreTask, luciTask)) {
+    /// Update [devicelabTask] when first [luciTask] run finishes. There may be
+    /// reruns for the same commit and same builder. Update [devicelabTask]
```

Comment:
What happens if there is a rerun? are they ignored?

Reply:
Suppose the first run with build number 123, then `builderNumberList='123'`.
When rerun happens with build number 456, the `builderNumberList` will be updated to `123,456`.

The status will be updated to the status of the new run of build 456 if changed.

With newly added doc. in `Task`, this should be easy to follow now.